tests/unit/test_util.py

```diff
@@ -266,6 +268,53 @@ def _mock(x=1.0, *, a=2.0):
         # `post_time` is upper bounded by `pre_time - wait_time`.
         assert post_time < (pre_time - wait_time)
 
+    def test_speed_comparison_test(self) -> None:
+        """
+        Check that ``speed_comparison_test`` returns expected timings.
+
+        ``speed_comparison_test`` returns raw timings and a list of summary statistics
+        for each passed function. The summary statistics include an  estimate of the
+        mean time for JIT compiling the passed function, and an estimate of the mean
+        time for executing the JIT compiled function.
+
+        In this test we check that the passed functions are compiled for each run and
+        not cached, and that a for-looped variant of a mean function takes longer to
+        compile than the in-built vectorised version.
+        """
+        trace_counter = Mock()
+        # Define a for-looped version of a mean computation, this will be very slow to
+        # compile.
+
+        def _slow_mean(a):
+            trace_counter()
+            num_points = a.shape[0]
+            total = 0
+            for i in range(num_points):
+                total += a[i]
+            return total / num_points
+
+        random_vector = jr.normal(jr.key(2_024), shape=(100,))
+
+        num_runs = 10
+        summary_stats, _ = speed_comparison_test(
```

Comment:
Keep the dictionary (second return) too. Let's do some more testing with it on slow mean:

1. Assert that all the uncompiled times are slower than precompiled.
2. Assert that the difference in means for `summary_stats` is (almost) equal to the mean of the dictionary values. [We can't do a similar thing for SD.]

Reply:
I am not sure if I am following exactly what these tests show?

Reply:
Might become clearer as we talk.